app/services/explorer.js

```diff
@@ -655,44 +657,42 @@ export default Ember.Service.extend({
      * @param {DS.Store} store
      * @return {Ember.RSVP.Promise<BucketList>} Result of the AJAX request
      */
-    getBucketList(cluster, bucketType, store) {
-        console.log('Refreshing buckets for bucketType: %O', bucketType);
-        var clusterId = cluster.get('clusterId');
-        var bucketTypeId = bucketType.get('bucketTypeId');
-        var url = this.apiURL + 'explore/clusters/' + clusterId +
-            '/bucket_types/' + bucketTypeId + '/buckets' ;
-        var explorer = this;
+    getBucketList(cluster, bucketType, store, start=1, rows=100) {
+        let explorer = this;
+        let clusterId = cluster.get('clusterId');
+        let bucketTypeId = bucketType.get('bucketTypeId');
+        let url = `${this.apiURL}explore/clusters/${clusterId}/bucket_types/${bucketTypeId}/buckets?start=${start}&rows=${rows}`;
 
         return new Ember.RSVP.Promise(function(resolve, reject) {
-            var ajaxHash = {
+            var xhrConfig = {
                 url: url,
                 dataType: 'json',
-                type: 'GET'
-            };
-            ajaxHash.success = function(data) { // Success, bucket list returned
-                console.log("Found bucket list");
-                bucketType.set('isBucketListLoaded', true);
-                resolve(explorer.createBucketList(data, cluster, bucketType, store));
-            };
-            ajaxHash.error = function(jqXHR, textStatus) {
-                // Fail (likely a 404, cache not yet created)
-                if(jqXHR.status === 404) {
-                    // Kick off a Cache Refresh, and repeat the getBucketList request
-                    console.log("kicking off cache refresh...");
-                    explorer.bucketCacheRefresh(clusterId, bucketTypeId);
-                    // Return an empty (Loading..) list. Controller will poll to
-                    // refresh it, later
-                    var emptyList = store.createRecord('bucket-list', {
-                        cluster: cluster,
-                        bucketType: bucketType
-                    });
-                    Ember.run(null, resolve, emptyList);
-                } else {
-                    Ember.run(null, reject, textStatus);
+                type: 'GET',
+                success: function(data) {
+                    data.maxBucketsPerRequest = rows; // Pass through what the chunk size should be
```

Comment:
This is how the default value that is set up above is passed through, it is attached to the ajax request data.